buildSrc/build.gradle.kts

```diff
@@ -0,0 +1,107 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+plugins {
+    id("org.jlleitschuh.gradle.ktlint") version "9.2.1"
+    id("com.github.sherter.google-java-format") version "0.9"
+    `kotlin-dsl`
+}
+
+repositories {
+    mavenLocal()
+    maven(url = "https://maven.google.com/")
+    mavenCentral()
+    maven(url = "https://storage.googleapis.com/android-ci/mvn/")
+    maven(url = "https://plugins.gradle.org/m2/")
+}
+
+// Refer latest "perf-plugin" released version on https://maven.google.com/web/index.html?q=perf-plugin#com.google.firebase:perf-plugin
+// The System property allows us to integrate with an unreleased version from https://bityl.co/3oYt.
+// Refer go/fireperf-plugin-test-on-head for more details.
+val perfPluginVersion = System.getenv("FIREBASE_PERF_PLUGIN_VERSION") ?: "1.4.1"
+
+googleJavaFormat(Action {
+    toolVersion = "1.10.0"
+})
+
+dependencies {
+    // Firebase performance plugin, it should be added here because of how gradle dependency
+    // resolution works, otherwise it breaks Fireperf Test Apps.
+    // See https://github.com/gradle/gradle/issues/12286
+    implementation("com.google.firebase:perf-plugin:$perfPluginVersion")
+
+    implementation("com.google.auto.value:auto-value-annotations:1.8.1")
+    annotationProcessor("com.google.auto.value:auto-value:1.6.5")
+    implementation("digital.wup:android-maven-publish:3.6.3")
+    implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.32")
+    implementation("org.json:json:20210307")
+
+    implementation("org.eclipse.aether:aether-api:1.0.0.v20140518")
+    implementation("org.eclipse.aether:aether-util:1.0.0.v20140518")
+    implementation("org.eclipse.aether:aether-impl:1.0.0.v20140518")
+    implementation("org.eclipse.aether:aether-connector-basic:1.0.0.v20140518")
+    implementation("org.eclipse.aether:aether-transport-file:1.0.0.v20140518")
+    implementation("org.eclipse.aether:aether-transport-http:1.0.0.v20140518")
+    implementation("org.eclipse.aether:aether-transport-wagon:1.0.0.v20140518")
+    implementation("org.apache.maven:maven-aether-provider:3.1.0")
+
+    implementation("com.google.code.gson:gson:2.8.9")
+    implementation("org.jetbrains.dokka:dokka-android-gradle-plugin:0.9.17-g013-9b8280a")
+    implementation("com.android.tools.build:gradle:3.4.3")
+    implementation("gradle.plugin.com.github.sherter.google-java-format:google-java-format-gradle-plugin:0.9")
+
+    testImplementation("junit:junit:4.13.2")
+    testImplementation("com.google.truth:truth:1.1.2")
+    testImplementation("commons-io:commons-io:2.6")
+}
+
+gradlePlugin(Action {
```

Comment:
Theres a gradle extension method on projects that has the action block as the last argument (actually, it's only argument), so Action shouldn't be needed here. That is, this should work:
```
gradlePlugin {
...
}
```

Reply:
While it works, studio fails to type check it, let's leave it as is for now

Reply:
> While it works, studio fails to type check it, let's leave it as is for now

Yeah its just a nit, so either way works. Although, I think that may be an issue with your AS cache- because mine works perfectly fine. Or maybe its a bug within AS, I'm currently on `2021.3.1 Beta 4`

Reply:
hm, it used to not work with AGP 7.2.2, but looks like it's not an issue anymore, fixed